lib/node_modules/@stdlib/array/base/to-inserted-at/lib/assign.js

```diff
@@ -0,0 +1,236 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isComplexDataType = require( '@stdlib/array/base/assert/is-complex-floating-point-data-type' );
+var isBooleanDataType = require( '@stdlib/array/base/assert/is-boolean-data-type' );
+var isComplexLike = require( '@stdlib/assert/is-complex-like' );
+var Boolean = require( '@stdlib/boolean/ctor' );
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+var reinterpret = require( '@stdlib/strided/base/reinterpret-complex' );
+var reinterpretBoolean = require( '@stdlib/strided/base/reinterpret-boolean' );
+var normalizeIndex = require( '@stdlib/ndarray/base/normalize-index' );
+var real = require( '@stdlib/complex/float64/real' );
+var imag = require( '@stdlib/complex/float64/imag' );
+var format = require( '@stdlib/string/format' );
+
+
+// FUNCTIONS //
+
+/**
+* Copies elements from one array to another array and inserts the element at the specified index to a provided value.
+*
+* @private
+* @param {Collection} x - input array
+* @param {integer} index - element index
+* @param {*} value - inserting value
+* @param {Collection} out - output array
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array
+*
+* @example
+* var x = [ 1, 2, 3, 4 ];
+*
+* var out = [ 0, 0, 0, 0, 0 ];
+* var arr = indexed( x, 0, 5, out, 1, 0 );
+* // returns [ 5, 1, 2, 3, 4 ]
+*
+* var bool = ( arr === out );
+* // returns true
+*/
+function indexed( x, index, value, out, stride, offset ) {
+	var io;
+	var i;
+
+	io = offset;
+	for ( i = 0; i < x.length; i++ ) {
```

Comment:
You need to rethink this algo. Currently, you copy the entire array then you copy on avg O(N/2) again in the loop below. Applies here and elsewhere.